Scripts/ParseEmailFiles/ParseEmailFiles.py

```diff
@@ -3307,10 +3320,64 @@ def handle_msg(file_path, file_name, parse_only_headers=False, max_depth=3):
     if not msg:
         raise Exception("Could not parse msg file!")
 
-    email_data = msg.as_dict(max_depth)
+    msg_dict = msg.as_dict(max_depth)
+    format_string = get_msg_mail_format(msg_dict)
+
+    headers = []
+    headers_map = dict()  # type: dict
+    header_key = 'initial key'
+    header_value = 'initial header'
+
+    for header in msg_dict['Headers'].split('\n'):
+        if is_valid_header_to_parse(header):
+            if not header[0] == ' ' and not header[0] == '\t':
+                if header_value != 'initial header':
+                    header_value = convert_to_unicode(header_value)
+                    headers.append(
+                        {
+                            'name': header_key,
+                            'value': header_value
+                        }
+                    )
+
+                    if header_key in headers_map:
+                        # in case there is already such header
+                        # then add that header value to value array
+                        if not isinstance(headers_map[header_key], list):
+                            # convert the existing value to array
+                            headers_map[header_key] = [headers_map[header_key]]
+
+                        # add the new value to the value array
+                        headers_map[header_key].append(header_value)
+                    else:
+                        headers_map[header_key] = header_value
+
+                header_words = header.split(' ')
+
+                header_key = header_words[0][:-1]
+                header_value = ' '.join(header_words[1:])
+                header_value = header_value[:-1] if header_value[-1:] == ' ' \
+                    else header_value
+
+            else:
+                header_value += header[:-1] if header[-1:] == ' ' else header
+
+    email_data = {
+        'To': msg_dict['To'],
+        'CC': msg_dict['CC'],
+        'From': msg_dict['From'],
+        'Subject': headers_map['Subject'],
+        'HTML': msg_dict['HTML'],
+        'Text': msg_dict['Text'],
+        'Headers': headers,
+        'HeadersMap': headers_map,
+        'Attachments': '',
+        'Format': format_string,
+        'Depth': MAX_DEPTH_CONST - max_depth
```

Comment:
are you sure this line is correct?

Reply:
Its the same as in line 3583, where the original version of this code was written